core/src/main/java/com/statful/client/core/StatfulClientImpl.java

```diff
@@ -232,49 +425,86 @@ public void forceSyncFlush() {
     }
 
     private SenderFacade put(final String metricName, final String value) {
-        SenderAPI metricsSenderAPI = MetricsSenderAPI.newInstance(this).with()
-                .configuration(configuration)
-                .aggregationFrequency(configuration.getDefaultAggregationFreq())
-                .name(metricName)
-                .value(value);
+        SenderAPI metricsSenderAPI = createPut(metricName, value, false, false)
+                .aggregationFrequency(configuration.getDefaultAggregationFreq());
 
         return new StatfulClientFacade(metricsSenderAPI);
     }
 
     private SenderFacade aggregatedPut(final String metricName, final String value, final Aggregation aggregation,
                                        final AggregationFrequency aggregationFrequency) {
-        SenderAPI metricsSenderAPI = MetricsSenderAPI.newInstance(this, true).with()
-                .configuration(configuration)
+
+        SenderAPI metricsSenderAPI = createPut(metricName, value, true, false)
                 .aggregation(aggregation)
-                .aggregationFrequency(aggregationFrequency)
+                .aggregationFrequency(aggregationFrequency);
+
+        return new StatfulClientFacade(metricsSenderAPI);
+    }
+
+    private SenderFacade sampledPut(final String metricName, final String value, final Integer sampleRate) {
+        SenderAPI metricsSenderAPI = createPut(metricName, value, false, true)
+                .aggregationFrequency(configuration.getDefaultAggregationFreq())
```

Comment:
For consistency with: 

`private SenderFacade put(final String metricName, final String value)`

I agree that this should be revised for both methods because it's confusing, but would prefer to do it on another PR.